src/engine/NamedResultCacheSerializer.h

```diff
@@ -0,0 +1,190 @@
+    serializer >> value;
+
+    // Use the store method to maintain consistency
+    store(key, std::move(value));
+  }
+}
+
+namespace ad_utility::serialization {
+
+// Serialization for `NamedResultCache::Value`
+// This serializes the complete Value including the `LocalVocab` with proper ID
+// remapping.
+AD_SERIALIZE_FUNCTION_WITH_CONSTRAINT(
+    (ad_utility::SimilarTo<T, NamedResultCache::Value>)) {
+  if constexpr (WriteSerializer<S>) {
+    // Serialize the LocalVocab first (required for ID remapping)
+    ad_utility::detail::serializeLocalVocab(serializer, arg.localVocab_);
+
+    // Serialize the IdTable (uses the `serializeIds` helper which handles
+    // LocalVocab IDs)
+    serializer << arg.result_->numRows();
+    serializer << arg.result_->numColumns();
+    for (const auto& col : arg.result_->getColumns()) {
+      // TODO<joka921> Discuss and fix the semantics of local vocabs when
+      // serializing IDs to an from the serializer.
+      AD_CORRECTNESS_CHECK(
+          ql::ranges::find(col, Datatype::LocalVocabIndex, &Id::getDatatype) ==
+              col.end(),
+          "Named result cache entries that contain local vocab entries "
+          "currently cannot be serialized. Note that local vocab entries can "
+          "also occur if SPARQL UPDATE operations have been performed on the "
+          "index before creating the named cached result.");
+      ad_utility::detail::serializeIds(serializer, col);
+    }
+
+    // Serialize VariableToColumnMap manually (can't use HashMap serialization
+    // because Variable is not default-constructible)
+    serializer << arg.varToColMap_.size();
+    for (const auto& [var, colInfo] : arg.varToColMap_) {
+      serializer << var;
+      serializer << colInfo;
+    }
+
+    // Serialize resultSortedOn (vector of ColumnIndex)
+    serializer << arg.resultSortedOn_;
+
+    // Serialize cacheKey (string)
+    serializer << arg.cacheKey_;
+
+    // Serialize cachedGeoIndex (optional)
+    // Note: We cannot serialize the `optional` directly, because the geo index
+    // has no default constructor for safety reasons.
+    bool hasGeoIndex = arg.cachedGeoIndex_.has_value();
+    serializer << hasGeoIndex;
+    if (hasGeoIndex) {
+      serializer << arg.cachedGeoIndex_.value();
+    }
+  } else {
+    // Deserialize the LocalVocab and get the ID mapping
+    AD_CORRECTNESS_CHECK(arg.blankNodeManagerForSerialization_.has_value());
+    auto [localVocab, mapping] = ad_utility::detail::deserializeLocalVocab(
+        serializer, &arg.blankNodeManagerForSerialization_.value());
+
+    // Deserialize the IdTable with ID mapping applied
+    size_t numRows, numColumns;
+    serializer >> numRows;
+    serializer >> numColumns;
+
+    AD_CORRECTNESS_CHECK(arg.allocatorForSerialization_.has_value());
+    IdTable idTable{numColumns, arg.allocatorForSerialization_.value()};
+    idTable.resize(numRows);
+    for (decltype(auto) col : idTable.getColumns()) {
```

Comment:
Isn't `auto&&` usually the thing to go for in this case? Or why is the `decltype` even required in the first place?